internal/cli/cmd/configurations.go

```diff
@@ -0,0 +1,363 @@
+// Copyright © 2021 - 2023 SUSE LLC
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+)
+
+//counterfeiter:generate -header ../../../LICENSE_HEADER . ConfigurationService
+type ConfigurationService interface {
+	Configurations(all bool) error
+	ConfigurationDetails(configuration string) error
+	CreateConfiguration(configuration string, kvAssigments []string) error
+	DeleteConfiguration(configurations []string, unbind bool, all bool) error
+	ConfigurationMatching(tocomplete string) []string
+	UpdateConfiguration(configuration string, removedKeys []string, assignments map[string]string) error
+	BindConfiguration(configuration, application string) error
+	UnbindConfiguration(configuration, application string) error
+
+	ConfigurationMatcher
+	ConfigurationAppMatcher
+}
+
+// NewConfigurationCmd returns a new 'epinio configuration' command
+func NewConfigurationCmd(client ConfigurationService, rootCfg *RootConfig) *cobra.Command {
+	configurationCmd := &cobra.Command{
+		Use:           "configuration",
+		Aliases:       []string{"configurations"},
+		Short:         "Epinio configuration features",
+		Long:          `Handle configuration features with Epinio`,
+		SilenceErrors: true,
+		SilenceUsage:  true,
+		Args:          cobra.MinimumNArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := cmd.Usage(); err != nil {
+				return err
+			}
+			return fmt.Errorf(`Unknown method "%s"`, args[0])
+		},
+	}
+
+	configurationCmd.AddCommand(
+		NewConfigurationBindCmd(client),
+		NewConfigurationCreateCmd(client),
+		NewConfigurationDeleteCmd(client),
+		NewConfigurationListCmd(client, rootCfg),
+		NewConfigurationShowCmd(client, rootCfg),
+		NewConfigurationUnbindCmd(client),
+		NewConfigurationUpdateCmd(client),
+	)
+
+	return configurationCmd
+}
+
+// NewConfigurationListCmd returns a new 'epinio configuration list' command
+func NewConfigurationListCmd(client ConfigurationService, rootCfg *RootConfig) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "list [--all]",
+		Short: "Lists configurations",
+		Long:  "Lists configurations in the targeted namespace, or all",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+
+			all, err := cmd.Flags().GetBool("all")
```

Comment:
Moving the other commands I started using structs instead of the `GetFoo()` func, with the `flag.BoolVar()`, instead of the `flag.Bool()`  
 I find it very verbose with many parameters, and when you have many flags that you need to pass to the underlying function if you have a "config struct" it's very convenient to pass everything and leave the logic to it.

I.e. for the [namespace delete command](https://github.com/epinio/epinio/blob/0d7c91219b59d20ed70645f192787f63e93ab566/internal/cli/cmd/namespaces.go#L97-L130) 

This is the Configuration struct:

https://github.com/epinio/epinio/blob/0d7c91219b59d20ed70645f192787f63e93ab566/internal/cli/cmd/namespaces.go#L97-L100

The flags are bound to the fields:

https://github.com/epinio/epinio/blob/0d7c91219b59d20ed70645f192787f63e93ab566/internal/cli/cmd/namespaces.go#L126-L127

and then you can just use them:

https://github.com/epinio/epinio/blob/0d7c91219b59d20ed70645f192787f63e93ab566/internal/cli/cmd/namespaces.go#L113

Reply:
It actually also removes quite a lot of error checks (cannot read flag this, that, or the other).

Reply:
done with commit 04b657595d4b19edb345884dfece5b2cb078ad03